src/search/srpso.c

```
@@ -0,0 +1,239 @@
+// cppcheck-suppress missingIncludeSystem
+#include <string.h>
+
+#include "search.h"
+#include "search_optimizer.h"
+#include "sr_alloc.h"
+#include "sr_rng.h"
+
+static int sr_pso_default_swarm_size(int ndim)
+{
+  const int base = 5 * ndim;
+  return (base < 10) ? 10 : base;
```

Clang (14.0.6)

[warning] 10 is a magic number; consider replacing it with a named constant

(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)